workorder-recheck.lua

```diff
@@ -3,18 +3,66 @@
 
 workorder-recheck
 =================
-Sets the status to ``Checking`` (from ``Active``) of the selected work order (in the ``j-m`` or ``u-m`` screens).
+Sets the status to ``Checking`` (from ``Active``) of the selected work order.
 This makes the manager reevaluate its conditions.
 ]====]
-local scr = dfhack.gui.getCurViewscreen()
-if df.viewscreen_jobmanagementst:is_instance(scr) then
-    local orders = df.global.world.manager_orders
-    local idx = scr.sel_idx
-    if idx < #orders then
-        orders[idx].status.active = false
+
+--@ module = true
+
+local gui = require('gui')
+local widgets = require('gui.widgets')
+local overlay = require('plugins.overlay')
+
+local function set_current_inactive()
+    local scrConditions = df.global.game.main_interface.info.work_orders.conditions
+    if scrConditions.open then
+        local order = scrConditions.wq
+        order.status.active = false
     else
-        qerror("Invalid order selected")
+        qerror("Order conditions is not open")
     end
-else
-    qerror('Must be called on the manager screen (j-m or u-m)')
 end
+
+-- -------------------
+-- RecheckOverlay
+--
+
+local focusString = 'dwarfmode/Info/WORK_ORDERS/Conditions'
+
+RecheckOverlay = defclass(RecheckOverlay, overlay.OverlayWidget)
+RecheckOverlay.ATTRS{
+    default_pos={x=6,y=2},
```

Comment:
we normally try to put the overlays on the panel itself or in the empty space to the right, but in this case those positions would be very far from the related vanilla status icon. I think the position you chose is probably the best choice, but we may get feedback from beta testers on this and end up moving it later, depending on how it interacts with vanilla elements on various screen sizes.